### PR TITLE
qemu_v8: build Xen with Buildroot

### PR DESCRIPTION
**Pull request by jforissier**
Commit 8b808915f856 ("qemu_v8: xen: apply patch to fix issue with RPC
buffer release") really did only one part of the job. Only the Xen tools
are built with Buildroot; the hypervisor itself (which needs the patch
mentioned in the commit) is actually built from the top-level xen/
directory. As a result the issue is stil present.
This commit changes the QEMUv8 makefiles as well as the Buildroot
configuration for Xen so that both the hypervisor and the tools are
built. As for the Xen configuration (enabling OP-TEE support), it is
introduced via a patch to the defconfig file. Patch file
"0001-Revert-xen-arm-Restrict-the-amount-of-memory-that-do.patch"
addresses a regression introduced in Xen 4.14.3 (which Buildroot
provides) compared to 4.14.1 (which is currently provided by
$(ROOT)/xen).

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jforissier** [comment]
@lorc what do you think of this "regression" I mentioned in the commit description? Is this something that should be addressed in OP-TEE or in Xen?

**Comment by jenswi-linaro** [comment]
Is there a risk of a chicken and egg problem with this? I mean buildroot depends on OP-TEE and it's not far fetched that some boot thing might depend on the Xen binary produced by buildroot.

**Comment by jforissier** [comment]
> Is there a risk of a chicken and egg problem with this? I mean buildroot depends on OP-TEE and it's not far fetched that some boot thing might depend on the Xen binary produced by buildroot.

Do you have anything specific in mind? The following builds and runs fine:
```
mkdir optee && cd optee
repo init -u https://github.com/OP-TEE/manifest.git -m qemu_v8.xml
repo sync -j20
rm -rf xen
cd build
git fetch github pull/565/head && git checkout FETCH_HEAD
make -j2 toolchains
make -j10 XEN_BOOT=y check
```

File: qemu_v8.mk [hunk 1]
```diff
@@ -374,46 +374,31 @@ $(ROOTFS_UGZ): u-boot buildroot | $(BINARIES_PATH)
...
-	$(MAKE) -C $(XEN_PATH)/xen XEN_TARGET_ARCH=arm64 defconfig
-	cd $(XEN_PATH)/xen && \
-	tools/kconfig/merge_config.sh -m .config $(ROOT)/build/kconfigs/xen.conf
-
-xen: $(XEN_PATH)/xen/.config
-	$(MAKE) -C $(XEN_PATH) dist-xen \
-	XEN_TARGET_ARCH=arm64 \
-	CONFIG_XEN_INSTALL_SUFFIX=.gz	\
-	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+#.PHONY: xen
```
Comment (jenswi-linaro) [inline qemu_v8.mk:377]: I guess this can be removed.

**Comment by jenswi-linaro** [comment]
> 

No, not really. I guess it's fine then.

File: br-ext/patches/xen/0001-Revert-xen-arm-Restrict-the-amount-of-memory-that-do.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
+From 82a8e6a916f1a7bb3910c365197319b37085d58c Mon Sep 17 00:00:00 2001
```
Comment (jenswi-linaro) [inline br-ext/patches/xen/0001-Revert-xen-arm-Restrict-the-amount-of-memory-that-do.patch:1]: Would it make sense to take this patch in a separate commit?

Comment (jenswi-linaro) [reply br-ext/patches/xen/0001-Revert-xen-arm-Restrict-the-amount-of-memory-that-do.patch:1]: Can't it be added before the "qemu_v8: build Xen with Buildroot" commit?

Comment (jforissier) [reply br-ext/patches/xen/0001-Revert-xen-arm-Restrict-the-amount-of-memory-that-do.patch:1]: Yes, I will split the commit.

**Comment by jforissier** [comment]
Update

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**